File: trendly_flutter/lib/models/comment.dart
class Comment {
  final int id;
  final int userId;
  final String userName;
  final String articleUrl;
  final String content;
  final DateTime? createdAt;

  Comment({
    required this.id,
    required this.userId,
    required this.userName,
    required this.articleUrl,
    required this.content,
    required this.createdAt,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'] is int ? json['id'] : int.parse(json['id'].toString()),
      userId: json['userId'] is int
          ? json['userId']
          : int.parse(json['userId'].toString()),
      userName: json['userName'] ?? '',
      articleUrl: json['articleUrl'] ?? '',
      content: json['content'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString())
          : null,
    );
  }
}

File: trendly-backend/src/main/java/com/trendly/trendlybackend/comment/model/Comment.java
package com.trendly.trendlybackend.comment.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "comments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // user who wrote the comment
    @Column(name = "user_id", nullable = false)
    private Long userId;

    // snapshot of user name for display (so we don't need join every time)
    @Column(name = "user_name", nullable = false)
    private String userName;

    // which article this comment belongs to
    @Column(name = "article_url", nullable = false, length = 1000)
    private String articleUrl;

    @Column(name = "content", nullable = false, length = 2000)
    private String content;

    private LocalDateTime createdAt;
}
